src/compiler/binder.ts

```diff
@@ -1104,6 +1104,9 @@ function createBinder(): (file: SourceFile, options: CompilerOptions) => void {
         // and set it before we descend into nodes that could actually be part of an assignment pattern.
         inAssignmentPattern = false;
         if (checkUnreachable(node)) {
+            if ((node as HasFlowNode).flowNode) {
+                (node as HasFlowNode).flowNode = undefined;
+            }

```

Comment:
```suggestion
            if (canHaveFlowNode(node)) {
                node.flowNode = undefined;
            }
```

To avoid casting, but also because I think it'll prevent negative perf effects from speculatively trying to access `flowNode` on all nodes?

Then again, the binder does not currently import this function at all, and it's not used below where `flowNode` is set, so maybe not.

Reply:
Calling `canHaveFlowNode` is better than the cast as it accesses a property that is always present (`kind`), vs. a property that may not be defined (`flowNode`). The latter results in eager deoptimizations as a result of the missing property.